### PR TITLE
OOP school library: decorate a class

### PR DESCRIPTION
**Pull request by Koshao-sys**
Hi,

I implemented the following requirements in this pull request;

**1. Interface**
- [ ] Create a class Nameable.
- [ ] Implement a method called correct_name that will raise a NotImplementedError.

**2. Turn your Person class to Nameable**
- [ ] Make sure that your Person class inherits from Nameable
- [ ] Make sure that this class has a method correct_name implemented. It should simply return the name attribute.

**3. Prepare base Decorator**
- [ ] Make sure that it inherits from Nameable.
- [ ] In the constructor assign a nameable object from params to an instance variable.
- [ ] Implement the correct_name method that returns the result of the correct_name method of the @nameable.

**4. Prepare CapitalizeDecorator and TrimmerDecorator**
- [ ] For the CapitalizeDecorator:
- Create a class that inherits from the base Decorator class.
- Implement a method correct_name that capitalizes the output of @nameable.correct_name.
- [ ] For the TrimmerDecorator:
- Create a class that inherits from the base Decorator class.
- Implement a method correct_name that makes sure that the output of @nameable.correct_name has a maximum of 10 characters. If it's longer it should trim the word.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  end
+
+  def can_use_services?
+    of_age? || @parent_permission
+  end
+
+  def of_age?
+    @age >= 18
+  end
+
```
Comment (DammyShittu) [inline person.rb]: - [ ] Kindly import the Nameable class here and make sure the Person class inherits from it.

  ```
  class Person < Nameable

- [ ] You are missing the `correct_name` method for the Person class

  ```
  def correct_name
    @name
  end
  ```

File: nameable.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class Nameable
+  def correct_name
+    raise 'NotImplementedError'
```
Comment (DammyShittu) [inline nameable.rb]: - [ ] Let's try to make the exception message more descriptive so we can also tell which class has failed to implement the `correct_name` method.

  ```
  raise NotImplementedError, "#{self.class} has not implemented method '#{__method__}'"
  ```

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @Koshao-sys,

## Highlights
✔️ Trimmer class

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Qoosim — APPROVED** [review]
#  *Status: Approved* :tada: :heavy_check_mark: 

Hi @Koshao-sys, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct Gitflow. :heavy_check_mark: 



## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**